### ISSUE TITLE
Why OD API caches groundtruth in a dict instead of passing to model.loss directly

### ISSUE THREAD

**Issue by djdongjin** [tensorflow/models#8980]
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [x] I checked to make sure that this feature has not been requested already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/object_detection

## 2. Describe the feature you request

I am trying to export a trainable SavedModel from TF2 object detection models. However, the training workflow involves saving groundtruth in a `dict` member (e.g. `self._groundtruth_dict`).

```python
model.provide_groundtruth(gt_dict)  # by calling provide_groundtruth(model, gt_dict) helper
pred = model.predict(...)
loss = model.loss(pred)
```

I think the `provide_groundtruth` step stops one from exporting a trainable SavedModel. Exporting an inference SavedModel doesn't have this issue since inference doesn't have this step. And I think this is the only factor that blocks me from exporting a trainable SavedModel.

Why not letting `provide_groundtruth(model, gt_dict)` return the transformed `gt_dict` directly and passing it to `model.loss`? We can still do some checking before passing the `gt_dict` to a model (might need to do nontrivial changes).

```python
gt_dict = provide_groundtruth(model, gt_dict)
pred = model.predict(...)
loss = model.loss(pred, gt_dict)
```

Or are there existing ways of exporting a trainable SavedModel that I'm not aware of.

Thanks!

## 4. Are you willing to contribute it? (Yes or No)

Yes.
